wp-content/themes/eightshift-boilerplate/src/class-config.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * The file that defines the main start class.
+ *
+ * A class definition that includes attributes and functions used across both the
+ * theme-facing side of the site and the admin area.
+ *
+ * @package Eightshift_Boilerplate\Core
+ */
+
+declare( strict_types=1 );
+
+namespace Eightshift_Boilerplate\Core;
+
+use Eightshift_Libs\Core\Config as Lib_Config;
+
+/**
+ * The project config class.
+ *
+ * @since 2.0.0
+ */
+class Config extends Lib_Config {
+
+  /**
+   * Method returns project name generally used for naming assets handlers, languages and etc.
+   *
+   * @since 2.0.0
+   */
+  public static function get_project_name() : string {
+    return 'eightshift-boilerplate';
+  }
+
+  /**
+   * Method returns project version generally used for versioning assets handlers.
+   *
+   * @since 2.0.0
+   */
+  public static function get_project_version() : string {
+    return '1.0.0';
+  }
+
+  /**
+   * Returns project prefix for adding it to all the filters as a prefix because all filters in WordPress live inside a global namespace.
+   *
+   * @return string Full path to asset.
+   *
+   * @since 2.0.0 Init
+   */
+  public static function get_project_prefix() : string {
+    return 'eb8';
+  }
+
+  /**
+   *
+   * Returns project env used to define global settings depending on the environment of the project.
+   *
+   * @return string Project env state.
+   *
+   * @since 2.0.0 Init
+   */
+  public static function get_project_env() : string {
+    return ES_ENV;
+  }
+
+  /**
+   * Method returns project primary color generally used for styling mobile browser color and splash screens. check head.php for details.
+   *
+   * @since 2.0.0
+   */
+  public static function get_project_primary_color() : string {
+    return '#900000';
+  }
+
+  /**
+   * Return project absolute path for theme use get_template_directory() and for plugin use __DIR__.
```

Comment:
```suggestion
   * Return project absolute path 
   *
   * If used in a theme use get_template_directory() and in case it's used in a plugin use __DIR__.
```

If this is going to be used for themes/plugins maybe we should do something like

```php
  /**
   * Get the possible locations for the view.
   *
   * @param string $uri URI of the view to get the locations for.
   *
   * @return array Array of possible locations.
   */
  protected function get_locations( string $path ) : array {
    return [
      trailingslashit( \get_stylesheet_directory() ) . $path,
      trailingslashit( \get_template_directory() ) . $path,
      trailingslashit( __DIR__ ) . $path,
    ];
  }
```

And then in the consuming code do something like

```php
foreach ( $this->get_locations( $path ) as $location ) {
  if ( is_readable( $location ) ) {
    return $location;
  }
}
```

This way we are covering multiple possibilities. Thoughts?

Reply:
Makes sense but I would like to test this. I have created an issue for this.
https://github.com/infinum/eightshift-libs/issues/20